resources/views/pages-legacy/viewtopic.blade.php

```diff
@@ -149,39 +155,35 @@
     }
     echo "</div>";
     echo "<div>";
-    RenderUpdateSubscriptionForm(
-        "updatetopicsubscription",
-        SubscriptionSubjectType::ForumTopic,
-        $thisTopicID,
-        $isSubscribed
-    );
+    if ($user) {
+        RenderUpdateSubscriptionForm(
+            "updatetopicsubscription",
+            SubscriptionSubjectType::ForumTopic,
+            $thisTopicID,
+            $isSubscribed
+        );
+    }
```

Comment:
This conditional fixes a subtle bug. Currently, for unauthenticated users, they still see the "Subscribe" button when viewing topics. Clicking it always returns an error. For now, we should probably just hide the button if the user isn't logged in.